dbt/adapters/dremio/api/rest/url_builder.py

```diff
@@ -31,92 +34,90 @@ class UrlBuilder:
     CLOUD_CATALOG_ENDPOINT = CLOUD_PROJECT_ENDPOINT + "/{}/catalog"
 
     @classmethod
-    def login_url(cls, base_url):
-        return base_url + UrlBuilder.SOFTWARE_LOGIN_ENDPOINT
+    def login_url(cls, parameters: Parameters):
+        return parameters.base_url + UrlBuilder.SOFTWARE_LOGIN_ENDPOINT
```

Comment:
Should we raise an error or warn if we get passed a CloudParameters object? If anything, that would provide documentation as to why there is no cloud login endpoint.

Reply:
Could the same goal be achieved by changing the method signature to `def login_url(cls, software_parameters: SoftwareParameters):`?

Reply:
Add comment why we're only accepting SoftwareParameters.